### PR TITLE
Add developer documentation

### PR DESCRIPTION
**Pull request by jdidion**

### CODE REVIEW

File: doc/developer.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    
+    ```
+    $ brew tap AdoptOpenJDK/openjdk
+    $ brew cask install adoptopenjdk8
+    # Use java_home to find the location of JAVA_HOME to set
+    $ /usr/libexec/java_home -V
+    $ export JAVA_HOME=/Library/Java/...
+    ```
+  * Scala will compile with JDK 11, but the JDK on DNAnexus worker instances is JDK 8 and will not be able to run a JAR file with classes compiled by a later version of Java
+* Install [sbt]() (e.g. `brew install sbt`), which also installs Scala
```
Comment (sstadick) [inline doc/developer.md]: link to https://www.scala-sbt.org/

File: doc/developer.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+$ find . -name target | xargs rm -rf
+```
+
+You may also need to delete artifiacts that have been cached on the platform:
+
+```
+$ dx rm -r dxWDL_playground:/builds/<username>/<version>
+```
+
+Next, you need to build the dxWDL JAR file. Run `sbt assembly`, then move the JAR file from the `target` folder to the root dxWDL folder, e.g. `mv target/dxWDL.jar ./dxWDL-1.44.jar`.
```
Comment (orodeh) [inline doc/developer.md]: This paragraph isn't needed. All you need to do is `./scripts/run_tests.py`

File: doc/developer.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    ```
+    $ brew tap AdoptOpenJDK/openjdk
+    $ brew cask install adoptopenjdk8
+    # Use java_home to find the location of JAVA_HOME to set
+    $ /usr/libexec/java_home -V
+    $ export JAVA_HOME=/Library/Java/...
+    ```
+  * Scala will compile with JDK 11, but the JDK on DNAnexus worker instances is JDK 8 and will not be able to run a JAR file with classes compiled by a later version of Java
+* Install [sbt]() (e.g. `brew install sbt`), which also installs Scala
+* We also recommend to install [Metals](), which enables better integration with your IDE
```
Comment (sstadick) [inline doc/developer.md]: link to https://scalameta.org/metals/

File: doc/developer.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  * For VSCode, install the "Scala (Metals)" and "Scala Syntax (official)" plugins
+* You will also need Python installed (we recommend version 3.6+) and dx-toolkit (`pip install dxpy`)
+
+## Getting the source code
+
+* Clone or fork the [dxWDL repository]() (depending on whether you have commit permissions)
```
Comment (sstadick) [inline doc/developer.md]: link to https://github.com/dnanexus/dxWDL

File: doc/developer.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+## Adding new code
+
+* Follow the style guidelines (below).
+* Always write unit tests for any new code you add, and update tests for any code you modify.
+* Submit a pull request when you are ready to have your code reviewed.
+
+### Style guidelines
+
+* We use [scalafmt style]() with a few modifications. You don't need to worry so much about code style since you will use the automatic formatter on your code before it is committed.
```
Comment (sstadick) [inline doc/developer.md]: link to https://scalameta.org/scalafmt/

**Comment by jdidion** [comment]
@sstadick thanks, there is a bit of overlap with Procedures. I will merge that document into developer.md. I'm not married to calling it "developer.md" - happy to take other suggestions, but I think it's purpose is a bit more clear than "Procedures".

**Comment by sstadick** [comment]
developer.md seems fine to me! I guess this is the kind of thing that would eventually roll into a larger wiki / full on docs site.

**Comment by jdidion** [comment]
Thanks both. I made all the suggested changes, and also merged in most things in "Procedures.md". The only thing remaining in that file is the build instructions, because I'm not sure if they're up to date. For example, they mention wdl4s, which is now called WOM. I think it's valuable to also have install/build instructions for unix, so I will merge them in after @orodeh checks to make sure everything is current.

**Comment by jtratner** [comment]
did developer.md get deleted?

**Comment by orodeh** [comment]
@jdidion I wanted to take a last look, and approve it, but I can't find the developer.md file.

**Comment by jdidion** [comment]
Ugh, sorry @orodeh @jtratner. I renamed the file but then forgot to git add it. It's there now (doc/Developing.md).

File: doc/Procedures.md [hunk 6]
```diff
@@ -1,34 +1,3 @@
...
-
-
-## Coding guidelines
-
-- Unit tests should assert, and not print to the console
-- WDL test files belong in the top directory `test`
-- Code should comply with scalafmt default settings + what is in the
-  .scalafmt.conf file at the top level of the project
-
 ## Building
```
Comment (orodeh) [inline doc/Procedures.md]: Just remove this paragraph. We no longer use dxjava, and `build_jar.py` should not be used manually.

Comment (jdidion) [reply doc/Procedures.md]: Done. I moved the remaining Linux instruction sinto Developing.md.

**Pull request merged**